src/amlaidatatests/tests/common.py

```diff
@@ -482,6 +482,85 @@ def _test(self, *, connection: BaseBackend) -> None:
             )
 
 
+class VerifyEntitySubset(AbstractColumnTest):
+    def __init__(
+        self,
+        *,
+        table_config: ResolvedTableConfig,
+        column: str,
+        superset_value: str,
+        subset_value: str,
+        concat: list[str],
+        test_id: Optional[str] = None,
+        severity: AMLAITestSeverity = AMLAITestSeverity.ERROR,
+    ) -> None:
+        """Checks for subset entity records without a corresponding superset record,
+        for cases where a superset record is a prerequisite for the existence of a
+        subset record.
+
+        Args:
+            table_config: The resolved table config to test
+            severity: The error type to emit on test failure
+                        Defaults to AMLAITestSeverity.ERROR
+            test_id: A unique identifier for the test
+            column: The column being tested
+            superset_value: the column value used to indicate a superset record
+            subset_value: the column value used to indicate a superset record
+            concat: list of column_ids to concat into a unique entity identifier
+            test_id: A unique identifier for the test. Defaults to None.
+            severity: The error type to emit on test failure
+                            Defaults to AMLAITestSeverity.WARN.
+        """
+        super().__init__(
+            table_config=table_config, column=column, severity=severity, test_id=test_id
+        )
+        self.superset_value = superset_value
+        self.subset_value = subset_value
+        self.concat = concat
+
+    def _test(self, *, connection: BaseBackend) -> None:
+        table = self.table
+        if self.table_config.table_type in (
+            TableType.CLOSED_ENDED_ENTITY,
+            TableType.OPEN_ENDED_ENTITY,
+        ):
+            # For these tables, we need to identify the latest row
+            table = self.get_latest_rows(table)
+        table, column = resolve_field(table=table, column=self.column)
+
+        group_concat = table.mutate(
+            concat=reduce(lambda x, y: x + y, [i + _[i] for i in self.concat], "")
+        )
+
+        subset_table = group_concat.filter(
+            group_concat[self.column] == self.subset_value
+        )
+
+        superset_table = group_concat.filter(
+            group_concat[self.column] == self.superset_value
+        )
+
+        joined = subset_table.left_join(
+            superset_table, predicates=[(subset_table.concat, superset_table.concat)]
+        )
+        anti_joined = joined.filter(joined.concat_right == None)
+
+        expr = anti_joined.aggregate(count=anti_joined.count())
+
+        results = connection.execute(expr).iloc[0]
+        count = results["count"]
+
+        if count > 0:
+            raise DataTestFailure(
```

Comment:
I agree that this would be a useful additional check. Do you think this should be included in the same common test? Might there be cases of the entity subset/superset relation that are not time dependent?

Reply:
There might well be cases like this. You could put an enum in as an order_check="<"/ ">"/"NONE"/">=","<=" or something like that.

Reply:
Discussed with @claw89 - won't do.